Amplify/Categories/DataStore/Model/Temporal/TemporalFormat.swift

```diff
@@ -21,6 +21,16 @@ public enum TemporalFormat: CaseIterable {
 
 extension TemporalFormat {
 
+    /// The order in which an implementation of `TemporalSpec` is
+    /// parsed matters so no precision is lost. This stored property
+    /// represents the order, from the least to the most precise format.
+    ///
+    /// - Note: if more formats are added to the enum, this property
+    /// needs to be updated to reflect the expected parsing order.
+    static var sortedCasesForParsing: [TemporalFormat] {
```

Comment:
Shouldn't this be in most-to-least precise, if we iterate over them in order as we attempt to parse?

Reply:
Good call-out. I think you're right. Swift `DateFormatter` is very strict when parsing so I wasn't able to reproduce a scenario where precision is lost. Regardless, the order you suggested you likely protected this API from such scenarios.

Reply:
Done!